src/Test/QuickCheck/Classes/Generic.hs

```diff
@@ -0,0 +1,119 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+#if MIN_VERSION_base(4,12,0)
+{-# LANGUAGE QuantifiedConstraints #-}
+#endif
+{-# OPTIONS_GHC -Wall #-}
+
+module Test.QuickCheck.Classes.Generic
+  (
+#if MIN_VERSION_base(4,6,0)
+    genericLaws
+  , genericLawsU
+#endif
+#if MIN_VERSION_base(4,12,0) 
+--  , generic1Laws
+#endif
+  ) where
+
+#if MIN_VERSION_base(4,6,0)
+import Control.Applicative
+import Data.Semigroup
+import Data.Monoid
+import GHC.Generics
+#if MIN_VERSION_base(4,9,0)
+import Data.Functor.Classes
+#endif
+import Data.Proxy (Proxy(Proxy))
+import Test.QuickCheck
+import Test.QuickCheck.Property (Property)
+
+import Test.QuickCheck.Classes.Common (Laws(..), Apply(..))
+
+-- | Tests the following properties:
+--
+-- [/Conjunction Idempotence/]
+--   @n .&. n ≡ n@
+-- [/Disjunction Idempotence/]
+--   @n .|. n ≡ n@
+-- [/Double Complement/]
+--   @complement (complement n) ≡ n@
+--
+-- /Note:/ This property test is only available when
+-- using @base-4.6@ or newer.
+genericLaws :: (Generic a, Eq a, Arbitrary a, Show a, Show (Rep a x), Arbitrary (Rep a x), Eq (Rep a x)) => Proxy a -> Proxy x -> Laws
+genericLaws pa px = Laws "Generic"
+  [ ("From-To idempotence", fromToIdempotence pa px)
```

Comment:
I don't think "idempotence" is the right word to describe this property, since idempotence usually refers to the property that some operation `*` satisfies `x * x = x`. I think "inverse" might be a better description.

Reply:
Yeah, this is correct. Will fix.